src/mantine-core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -345,6 +345,45 @@ export const NumberInput = forwardRef<HTMLInputElement, NumberInputProps>(
       }
     };
 
+    const getMobileOperatingSystem = () => {
+      const userAgent = navigator.userAgent || navigator.vendor;
```

Comment:
This will break during ssr, this code should only run in browser. Check if `typeof window !== 'undefined'` before running any browser specific code.